### ISSUE TITLE
Next.js 14 Conflict: Combining Server and Client Providers (i18n and AuthProvider)

### ISSUE THREAD

**Issue by buraxta** [refinedev/refine#6567]
### Describe the bug

When attempting to integrate both internationalization (i18n) and authentication providers in a Next.js 14 application using server and client components, I'm encountering a configuration conflict. 

### Steps To Reproduce

Create a Next.js 14 application with refine
Add internationalization provider
Add authentication provider
Attempt to use both in the same application layout or context
![image](https://github.com/user-attachments/assets/e8e6a1d0-2673-4791-b6a6-0f3e2b05e8f5)
![image](https://github.com/user-attachments/assets/18dabe9e-724b-4c6e-8dcb-b003cc0bf39d)

### Expected behavior

Smooth integration of server-side and client-side providers with seamless authentication and internationalization.

### Packages

same as this project: https://github.com/refinedev/refine/tree/master/examples/i18n-nextjs

### Additional Context

**Potential Solutions Attempted**

Separating providers into server and client components
Using server actions
Conditional rendering
Wrapper components

**Comment by alicanerdurmaz** [comment]
Hello @buraxta, thanks for the issue.

I added an auth-provider to our [i18n-nextjs](https://github.com/refinedev/refine/tree/main/examples/i18n-nextjs) example and fixed the issue [in this commit](https://github.com/refinedev/refine/pull/6571/commits).


Before the fix, we had a structure like this:
```
└── 📁auth-provider
    └── auth-provider.server.ts
    └── auth-provider.ts
    └── index.ts
```
Having both providers exported from `index.ts` was causing confusion for the Next.js app.

I refactored it as follows:
```
└── 📁auth-provider
    └── auth-provider.client.ts
    └── auth-provider.server.ts
```

And I renamed the authProvider objects to `authProviderClient` and `authProviderServer` to **minimize** confusion.

Now, when imported from the appropriate paths where needed, there seems to be no issue:
```ts
import { authProviderServer } from "@providers/auth-provider/auth-provider.server";
import { authProviderClient } from "@providers/auth-provider/auth-provider.client";
```

Thanks again for reporting the issue. If you encounter any other problems, please let us know 🚀

**Comment by buraxta** [comment]
Hello @refinedev team,

I cloned the i18n-nextjs repository from GitHub, but I encountered issues when trying to install the packages. For example, I got the following error for one of the packages:

```
npm error code ETARGET  
npm error notarget No matching version found for @refinedev/devtools@^1.2.11.  
npm error notarget In most cases you or one of your dependencies are requesting  
npm error notarget a package version that doesn't exist.  
```
I experienced similar errors for other packages as well. I had to downgrade the package versions to their previous versions to make it work.

I just wanted to share this additional information. Thank you in advance for your quick response and support! 😊
